engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java

```diff
@@ -154,7 +154,8 @@ public String onEntry(final String targetKey, final Expression sourceExpression)
         final String expression;
 
         if (sourceExpression instanceof StaticExpression) {
-          expression = String.format("\"%s\"", sourceExpression.getExpression());
+          expression =
+              String.format("\"%s\"", sourceExpression.getExpression().replaceAll("\"", "\\\\\""));
```

Comment:
:wrench: Please add a comment that explains what is happening here and why. :sweat_smile:

Reply:
We don't know what is happening here 🤣